File: MegaMemo/Scripts/Custom/card.js
var Card = function (deckId, front, back, reverse) {
    var self = this;
    self.data = {
        deckId: deckId,
        front: front,
        back: back,
        
        repetitionCount: 0,
        easinessFactor: 2.5,
        nextRepetitionDate: Date.now(),
        daysToNextRepetition: 0
    };
};

Card.makeReview = function (data, grade) {
    if (grade < 3) {
        data.repetitionCount = 0;
        data.nextRepetitionDate = Date.now();
        data.daysToNextRepetition = 0;
    }
    else {
        ++data.repetitionCount;

        if (data.repetitionCount == 1)
            data.daysToNextRepetition = 1;
        else if (data.repetitionCount == 2)
            data.daysToNextRepetition = 6;
        else
            data.daysToNextRepetition = parseInt(data.daysToNextRepetition * data.easinessFactor);

        daysToNextRepetition = data.daysToNextRepetition;
        data.nextRepetitionDate = moment(data.nextRepetitionDate).add('days', data.daysToNextRepetition).valueOf();
    }

    data.easinessFactor = data.easinessFactor + (0.1 - (5 - grade) * (0.08 + (5 - grade) * 0.02));
 
    if (data.easinessFactor < 1.3)
        data.easinessFactor = 1.3;
};

File: MegaMemo/Models/Card.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace MegaMemo.Models
{
    public class Card
    {
        [JsonIgnore]
        [DatabaseGeneratedAttribute(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [JsonIgnore]
        public int UserId { get; set; }

        [JsonProperty("id")]
        public int ClientId { get; set; }

        [JsonProperty("lastUpdateDate")]
        public long LastUpdateDate { get; set; }

        [JsonProperty("deckId")]
        public int DeckId { get; set; }

        [JsonProperty("front")]
        public string Front { get; set; }

        [JsonProperty("back")]
        public string Back { get; set; }

        [JsonProperty("repetitionCount")]
        public int RepetitionCount { get; set; }

        [JsonProperty("easinessFactor")]
        public double EasinessFactor { get; set; }

        [JsonProperty("nextRepetitionDate")]
        public long NextRepetitionDate { get; set; }

        [JsonProperty("daysToNextRepetition")]
        public int DaysToNextRepetition { get; set; }
    }
}